src/utils/emulators/pseudoElementFactory.js

```diff
@@ -0,0 +1,89 @@
+var List = require('csso:utils/list.js');
+var walk = require('utils.walker');
+var translate = require('csso:utils/translate.js');
+var pseudoContentFactory = require('./contentFactory');
+
+module.exports = function pseudoElementFactory(type, before) {
+  var TYPE_NAME = 'pseudo-element-' + type + '__' + basis.genUID();
+
+  return {
+    getStates: function() {
+      return [];
+    },
+    handleToken: function(token, parent, root, sourceMap) {
+      if (token.type == 'PseudoElement' && token.name == type) {
+        token.type = 'Combinator';
+        token.name = ' > ';
+        parent.data.sequence.appendList(new List([{type: 'Identifier', name: TYPE_NAME}]));
+      }
+    },
+    emulate: function(token, parent, root, sourceMap, mapper, value) {
+      var needToRetranslate = false;
+      var sourceToken = sourceMap.get(token);
+
+      if (sourceToken && sourceToken.type == 'PseudoElement' && sourceToken.name == type) {
+        var elementHandler;
+        var allowToEmulate = [];
+        var mappedElements = mapper.bySelector(token);
+
+        if (mappedElements) {
+          allowToEmulate = mappedElements.filter(function(element) {
+            var list = new List(parent.data.sequence.toArray().slice(0, -2));
```

Comment:
pseudo elements don't generates pseudo-states and should be applied only if they really neeed.
e.g.; `.button:hover .caption:active::before { ... }`
before-emulator should be applied only if `hover` and `active` emulators is applied
that's why we should understand - do we need to apply pseudo-element-emulator for this case or not
we do `querySelector` for understand - have we an element that matches that selector or not
`-2` because pseudo element emulation is transforming
`.button:hover .caption:active::before`
to
`.button:hover .caption:active > pseudo-element-before__fk43t23jh2jh`
but we should pass `.button:hover .caption:active` to `querySelector`
thats why we slice till `-2`